tests/user_restricted/test_user_restricted.py

```diff
@@ -82,13 +84,48 @@ def test_retrieve_patient_with_invalid_auth_header(self, headers):
         helpers.check_response_status_code(response, 401)
         helpers.check_response_headers(response, headers)
 
-    def test_retrieve_patient_with_missing_urid_header(self, headers_with_token):
+    def test_user_role_sharedflow_retrieve_patient_with_missing_urid_header(self, headers_with_token):
         self.headers.pop("NHSD-Session-URID")
         response = helpers.retrieve_patient(
-            retrieve[3]["patient"],
+            retrieve[0]["patient"],
             self.headers
         )
-        helpers.check_retrieve_response_body(response, retrieve[3]["response"])
+        helpers.check_response_headers(response, self.headers)
```

Comment:
Can we seperate this response shape check logic into a seperate helper function as its duplicated within the test_retrieve_patient test also?

Reply:
moved reposne body shape check into seperate helper func for reusability